src/renamewheel/main.py

```diff
@@ -1,86 +1,149 @@
 from auditwheel.wheeltools import get_wheel_architecture, get_wheel_libc
-from packaging.utils import InvalidWheelFilename
+from packaging.utils import InvalidWheelFilename, parse_wheel_filename
 
+# --- Constants for Exit Codes ---
+EXIT_SUCCESS = 0
+EXIT_WRONG_PLATFORM = 1
+EXIT_FILE_NOT_FOUND = 2
+EXIT_NOT_PLATFORM_WHEEL = 3
+EXIT_OUTPUT_DIRECTORY_NOT_FOUND = 4
+
+
+# --- Custom Exceptions ---
+class RenameWheelError(Exception):
+    """Base exception for this script."""
+
+
+class WheelNotFoundError(RenameWheelError):
+    """Raised when the wheel file does not exist."""
+
+
+class NotPlatformWheelError(RenameWheelError):
+    """Raised when the file is not a valid platform wheel."""
 
-def _parse_args():
-    p = argparse.ArgumentParser(description="Rename Linux Python wheels.")
-    p.add_argument("WHEEL_FILE", help="Path to wheel file.")
-    p.add_argument("-v", "--verbose", action="store_true", help="Print out messages to the console.")
-    p.add_argument("-w", "--working-dir", help="Copy file to working directory")
 
+def _parse_args() -> argparse.Namespace:
+    """Parses command-line arguments using pathlib for paths."""
+    p = argparse.ArgumentParser(description="Rename Linux Python wheels based on their ABI tag.")
+    p.add_argument("WHEEL_FILE", type=pathlib.Path, help="Path to wheel file.")
+    p.add_argument("-v", "--verbose", action="store_true", help="Print out messages to the console.")
+    p.add_argument("-w", "--working-dir", type=pathlib.Path, help="Copy the renamed wheel to this directory.")
     return p.parse_args()
 
 
-def _analyse_wheel(wheel_file, verbose):
-    if not isfile(wheel_file):
-        if verbose:
-            print(f"Cannot access {wheel_file!r}. No such file.")
-        return 2
+def _analyse_wheel_abi_tag(wheel_path: pathlib.Path) -> str:
+    """
+    Analyzes a wheel file to determine its platform ABI tag.
 
-    try:
-        arch = get_wheel_architecture(wheel_file)
-    except (WheelToolsError, NonPlatformWheel):
-        arch = None
+    Args:
+        wheel_path: The path to the wheel file.
 
-    try:
-        libc = get_wheel_libc(wheel_file)
-    except WheelToolsError:
-        libc = None
+    Returns:
+        The determined platform ABI tag (e.g., 'manylinux_2_17_x86_64').
 
-    if arch is None or libc is None:
-        if verbose:
-            print(f"This does not look like a platform wheel {wheel_file!r}.")
-        return 3
+    Raises:
+        WheelNotFoundError: If the wheel_path does not point to a file.
+        NotPlatformWheelError: If the file is not a valid platform wheel.
+    """
+    if not wheel_path.is_file():
+        raise WheelNotFoundError(f"Cannot access '{wheel_path}'. No such file.")
 
-    winfo = analyze_wheel_abi(libc, arch, pathlib.Path(wheel_file), frozenset(), True, True)
+    try:
+        # auditwheel functions expect string paths
+        str_path = str(wheel_path)
+        captured_error = StringIO()
+        with redirect_stderr(captured_error):
```

Comment:
Suppressing error messages seems like a bad idea?

Reply:
Catching messages from the inner library that the library is propagating itself. Wanting to control the output to the terminal.  It is potentially a bad idea, maybe another flag to turn it off for debugging if things aren't going as expected or parcel it with the change to quiet flag.

Reply:
I don't understand why you would ever want to suppress any output from this tool.